react/ui-components/src/hoc/app-config-screen-builder/AppFieldComposer.js

```diff
@@ -0,0 +1,1173 @@
+      }}
+      className="appConfigLabelField-toggleVisibility"
+    >
+      <Switch label="" isCheckedInitially={config?.hidden === false} />
+    </div>
+  );
+};
+
+// Component to render a delete button (dustbin icon) if deletion is allowed
+const DeleteFieldControl = ({ isDelete, onDelete }) => {
+  if (!isDelete) return null;
+
+  return (
+    <div
+      onClick={(e) => {
+        e.stopPropagation();
+        onDelete();
+      }}
+      style={{
+        cursor: "pointer",
+        fontWeight: "600",
+        marginLeft: "1rem",
+        fontSize: "1rem",
+        color: PRIMARY_COLOR,
+        display: "flex",
+        gap: "0.5rem",
+        alignItems: "center",
+        marginTop: "1rem",
+      }}
+    >
+      <DustbinIcon />
+    </div>
+  );
+};
+
+// Main component to display a panel field with label, tag, visibility toggle, and delete option
+const PanelFieldDisplay = ({ t, label, appType, config, onToggle, isDelete, onDelete }) => {
+  return (
+    <>
+      <div className="appConfigLabelField-label-container">
+        <div className="appConfigLabelField-label">
+          <span>{t(label)}</span>
+        </div>
+        <Tag icon="" label={t(appType)} className="app-config-field-tag" labelStyle={{}} showIcon={false} style={{}} />
+      </div>
+
+      {/* Control to show/hide the field */}
+      <ToggleVisibilityControl config={config} onToggle={onToggle} />
+
+      {/* Control to delete the field */}
+      <DeleteFieldControl isDelete={isDelete} onDelete={onDelete} />
+    </>
+  );
+};
+
+const MdmsDropdown = ({
+  t,
+  moduleMaster,
+  optionKey = "code",
+  moduleName,
+  masterName,
+  className,
+  style = {},
+  variant = "",
+  selected,
+  select = () => {},
+  rest,
+}) => {
+  if (!moduleName || !masterName) return null;
+  const { isLoading, data } = Digit.Hooks.useCustomMDMS(
+    Digit.ULBService.getCurrentTenantId(),
+    moduleName,
+    [{ name: masterName }],
+    {
+      enabled: moduleName && masterName,
+      select: (data) => {
+        return data?.[moduleName]?.[masterName]?.filter((item) => item.active);
+      },
+    },
+    { schemaCode: "MDMSDROPDOWNLIST" } //mdmsv2
+  );
+
+  if (isLoading) return <div>Loading...</div>;
+  return (
+    <Dropdown
+      className={className}
+      style={style}
+      variant={variant}
+      t={t}
+      option={data}
+      optionKey={optionKey}
+      selected={selected}
+      select={() => select()}
+    />
+  );
+};
+
+const Field = ({
+  t,
+  isDrawer = true,
+  headerFields,
+  type,
+  label,
+  value,
+  required,
+  isDelete,
+  onDelete,
+  onSelectField,
+  dropDownOptions,
+  config,
+  onChange,
+  onHide,
+  state,
+  dispatch,
+  Mandatory,
+  helpText,
+  infoText,
+  innerLabel,
+  rest,
+  ...props
+}) => {
+  switch (type) {
+    case "textarea": {
+      const [textVal, setTextVal] = useState(null);
+
+      useEffect(() => {
+        setTextVal(value);
+      }, [value]);
+      return (
+        <>
+          {infoText && (
```

Comment:
also just see if you can make a common between all these cases into one wrapper component